prov/efa/src/rxr/rxr_ep.c

```diff
@@ -2502,29 +2501,26 @@ int rxr_ep_alloc_app_device_info(struct fi_info **app_device_info,
 }
 
 /**
- * @brief Create ibv_cq_ex by calling ibv_create_cq_ex
+ * @brief Set rxr_ep->ibv_cq_ex and rxr_ep->ibv_cq_ex_type by calling
+ * efadv_create_cq or ibv_create_cq_ex
  *
- * @param[in] cq Pointer to the efa_cq.
- * @param[in] attr Pointer to fi_cq_attr.
- * @return Return pointer to ibv_cq_ex if successful, otherwise NULL
+ * @param[in,out] ep Pointer to rxr_ep
+ * @param[in] attr Pointer to fi_cq_attr
+ * @return Return 0 on success, error code otherwise
  */
-static inline struct ibv_cq_ex *rxr_ep_create_ibv_cq_ex(struct rxr_ep *ep, struct fi_cq_attr *attr)
+static inline int rxr_ep_create_ibv_cq_ex(struct rxr_ep *ep, struct fi_cq_attr *attr)
 {
-	struct ibv_cq_init_attr_ex init_attr_ex = {
-		.cqe = attr->size ? attr->size : EFA_DEF_CQ_SIZE,
-		.cq_context = NULL,
-		.channel = NULL,
-		.comp_vector = 0,
-		/* EFA requires these values for wc_flags and comp_mask.
-		 * See `efa_create_cq_ex` in rdma-core.
-		 */
-		.wc_flags = IBV_WC_STANDARD_FLAGS,
-		.comp_mask = 0,
-	};
-
 	struct efa_domain *efa_domain = rxr_ep_domain(ep);
 
-	return ibv_create_cq_ex(efa_domain->device->ibv_ctx, &init_attr_ex);
+	/*
+	 * Set ep->ibv_cq_ex_type to IBV_CQ by default
+	 * If efa_cq_ibv_cq_ex_open succeeds in using efadv_create_cq,
+	 * then it will update ep->ibv_cq_ex_type to EFADV_CQ
+	 */
+	ep->ibv_cq_ex_type = IBV_CQ;
```

Comment:
This assignment is already done  in `efa_cq_ibv_cq_ex_open` right?  Can you remove it here?

Reply:
If I remove this, `ep->ibv_cq_ex_type` will be uninitialized. I need a null check later on, so leaving `ep->ibv_cq_ex_type` uninitialized could cause that null check to fail

Reply:
Talked offline. No need for null check because the address is never null, fixed